### PR TITLE
fix: On 304 for SSE - reconnect when not connected

### PR DESCRIPTION
**Pull request by JamieSinn**

### CODE REVIEW

File: configmanager.go [hunk 1]
```diff
@@ -251,6 +251,20 @@ func (e *EnvironmentConfigManager) fetchConfig(numRetriesRemaining int, minimumL
 		resp.Request = req
 		return e.setConfigFromResponse(resp)
 	case statusCode == http.StatusNotModified:
+		if e.sseManager != nil && !e.sseManager.Connected.Load() && e.minimalConfig != nil && e.minimalConfig.SSE != nil {
+			configUpdatedEvent := api.ClientEvent{
```
Comment (jonathannorris) [inline configmanager.go:255]: is this a common event? should it be in a shared function somewhere?

Comment (JamieSinn) [reply configmanager.go:255]: Not really - this is specific to the event handler flow

Comment (jonathannorris) [reply configmanager.go:255]: how is this triggered for the 200 responses?

Comment (JamieSinn) [reply configmanager.go:255]: Via the set config event; this one is specific because it's passing an existing property back - not a new url

**Review by jonathannorris — APPROVED** [review]

File: configmanager.go (hunk 1 above)
Comment (JamieSinn) [reply configmanager.go:255]: https://github.com/DevCycleHQ/go-server-sdk/blob/main/configmanager.go#L318

**Pull request merged**